src/Core/Search/Builder/TypedBuilder/ProductCombinationFiltersBuilder.php

```diff
@@ -65,12 +66,9 @@ public function buildFilters(Filters $filters = null)
         }
 
         $productId = $this->getProductId();
-        $shopId = $this->getShopId();
-        $filterId = ProductCombinationFilters::generateFilterId($productId, $shopId);
         $filterParameters['filters']['product_id'] = $productId;
-        $filterParameters['filters']['shop_id'] = $shopId;
 
-        return new ProductCombinationFilters($filterParameters, $filterId);
+        return new ProductCombinationFilters($this->buildShopConstraint(), $filterParameters);
```

Comment:
Actually, since you extend `AbstractFiltersBuilder` you should be able to sue the protected field `$this->shopConstraint` directly, which is based on the provided shopConstraint in the builder's process

It is correctly filled here:
https://github.com/PrestaShop/PrestaShop/blob/efa87169070319d565c5a665d2add7e01778abfc/src/PrestaShopBundle/Controller/ArgumentResolver/FiltersBuilderResolver.php#L67-L76

for all potential builders, and it's correctly fetched from the new request attributes storing the shop constrait. It's better to rely on this one since it's supposed to replace the ShopContext in the long term (we'll be able to control the shop context via GET parameters via this some time in the future)

Reply:
but it will make the shopId coming from request useless now as it still takes shop constraint from context